Added minimum and maximum player settings and a help command

qualitykk opened the pull request:
- Settings save to a json file
- Added help command so people know what the bot can do
- In depth help per command also works
- No more connection verification spam by default

(Idk why this includes all the edits i made for the c!play arguments thing but it does so idk)

qualitykk committed:
Added using arguments to pass provider, region and maps instead of reactions

- Added name -> provider and name -> region translation dictionaries

qualitykk committed:
Fixes

- moved crucial variables out of if statements
- Added myself to the credits :sunglasses:

qualitykk committed:
Oops

qualitykk committed:
Now edits instead of reposting to avoid ghost messages

qualitykk committed:
Added per-user setting support

- current settings: max_players and min_players
- added help command
- added UserData class to store the persistent information

qualitykk committed:
Fixes, fixes and more fixes

- added a command to list your current settings
- Moved default settings into a global variable
- moved UserData.json into /data to make it clearer that it isnt a configuration file
- fixed json.dump missing the file and writing perms
- added debug launch parameter which stops printing every time a server connection is verified
- Fixes

zonical reviewed (changes requested):
Good start so far. I'd recommend adding more settings such as default region(s) and default provider into the settings list. Additional addition: Can you add a confirmation prompt asking whether the user wants to use their default settings for their search or use custom ones? If you think it doesn't work, feel free to remove it.

qualitykk commented:
> Good start so far. I'd recommend adding more settings such as default region(s) and default provider into the settings list. Additional addition: Can you add a confirmation prompt asking whether the user wants to use their default settings for their search or use custom ones? If you think it doesn't work, feel free to remove it.

Will do

qualitykk committed:
Added provider and region setting

- Added default provider and default region setting
- calling c!play without arguments and default settings now asks for confirmation
- Added GetOrRegisterUser helper function incase we dont want to handle non registered users

qualitykk committed:
oops again

- fuck you vscode

qualitykk commented:
Changes have been made, it should now be ready to push

Commit:
Merge branch 'master' into master
